Send an alert in AdminChat whenever a TNT gets defused

qlados opened the pull request:
Closes #557 
Whenever an in-game player defuses a griefer's TNT, a message will be sent in the AdminChat. The formatting looks like this:

![alert](https://user-images.githubusercontent.com/25369949/87203622-4f3df480-c303-11ea-9db9-56f594036355.png)

qlados committed:
Broadcast AdminChat whenever TNT is defused

Signed-off-by: Valky <vicnicolas200@gmail.com>

qlados committed:
Fix the formatting

Signed-off-by: Valky <vicnicolas200@gmail.com>

qlados committed:
Make the chat color light red

Signed-off-by: Valky <vicnicolas200@gmail.com>

AustinLMayes commented:
You need to sign all of your commits and run the maven format utility (`mvn com.coveo:fmt-maven-plugin:format`)

Please remember to do this before making any future pull requests so that the checks will pass.

CoWinkKeyDinkInc commented:
Yes, there is a way to sign it retroactively I but it was too frustrating for me to do so I had to do the whole process over again.

qlados committed:
Fix formatting errors

Signed-off-by: Valky <vicnicolas200@gmail.com>

I forgot to run the maven formatting command, so this commit was needed as well.

qlados commented:
All commits have been signed off. Sorry for the inconveniences :$

EricZeiberg commented on util/src/main/i18n/templates/moderation.properties:
```diff
@@ -150,6 +150,10 @@ moderation.defuse.enemy = You may not defuse enemy TNT.
 
+# {0} = defuser player name
+# {1} = griefer player name
+moderation.defuse.message = {0} defused {1}'s TNT!
```
I would remove the exclamation mark. Don't think any other messages have that

AustinLMayes replied:
Look down like 4 translations in the file. It's pretty uncommon to not use them, tough.

EricZeiberg replied:
Alright, fair enough. Still believe that we should remove all exclamation points cause it looks unprofessional, but that's a discussion for another time.

BGMP replied:
I like it with the exclamation mark ❗

Electroid replied:
Translations should avoid punctuation, so let's not use the "!"

Pablete1234 replied:
Why would translations explicitly not use punctuation?

qlados replied:
I removed the exclamation mark from the message

KingOfSquares commented on util/src/main/i18n/templates/moderation.properties:
```diff
@@ -150,6 +150,10 @@ moderation.defuse.enemy = You may not defuse enemy TNT.
 moderation.defuse.displayName = TNT Defuser
 
+# {0} = defuser player name
+# {1} = griefer player name
```
`{1} = TnT owner player name`? Just not `griefer` imo

qlados replied:
Changed it to defusing and griefing player name

KingOfSquares commented on core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java:
```diff
@@ -90,6 +90,13 @@ private void participantDefuse(Player player, Entity entity) {
+                "moderation.defuse.message",
+                TextColor.RED,
+                clicker.getName(NameStyle.FANCY),
+                owner.getName(NameStyle.FANCY)),
```
`owner` is annotated as `@Nullable` so you need to account for that

qlados replied:
If the owner is found to be null, a seperate message including the defusing players name is being sent now

Pugzy commented:
I would try follow the same color and wording of the [OCN version](https://github.com/OvercastNetwork/ProjectAres/blob/b883ef579938c49076379fa306592da847fc672f/PGM/src/main/java/tc/oc/pgm/antigrief/DefuseListener.java#L95).

qlados committed:
Removed the exclamation mark and renamed the players

Signed-off-by: Valky <vicnicolas200@gmail.com>

qlados committed:
Accounting for owner == null case

Signed-off-by: Valky <vicnicolas200@gmail.com>

KingOfSquares commented:
When you are done with this(if you are going to change colors) would you mind post a screenshot of how these messages both look in-game? Going to use them for Crowdin

Pugzy commented on util/src/main/i18n/templates/moderation.properties:
```diff
@@ -154,6 +154,9 @@ moderation.defuse.displayName = TNT Defuser
 moderation.defuse.message = {0} defused {1}'s TNT
 
+# {0} = defusing player name
+moderation.defuse.message.unknown = {0} defused a TNT
```
Can players defuse TNT that doesn't belong to a teammate?

I'm not sure if moderators need notifying of this as its not team grief.

qlados replied:
I'm pretty certain they cannot. That message is for the case that the griefing player leaves and is accounted as null.

qlados replied:
Covering this review: https://github.com/PGMDev/PGM/pull/559#discussion_r453207526

qlados committed:
Re-color the message

Signed-off-by: Valky <vicnicolas200@gmail.com>

qlados commented:
Looks now like the old OCN version

![messages](https://user-images.githubusercontent.com/25369949/87341445-ef855a80-c549-11ea-9a22-0cd1b7745f81.png)

Pablete1234 commented on core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java:
```diff
@@ -90,12 +91,27 @@ private void participantDefuse(Player player, Entity entity) {
+                "moderation.defuse.message.unknown",
+                TextColor.WHITE,
+                clicker.getName(NameStyle.FANCY),
+                TextComponent.of("TNT", TextColor.DARK_RED)),
```
Needs to be a translated string, present in the minecraft translations

Pablete1234 commented on core/src/main/java/tc/oc/pgm/listeners/AntiGriefListener.java:
```diff
@@ -90,12 +91,27 @@ private void participantDefuse(Player player, Entity entity) {
+                TextColor.WHITE,
+                clicker.getName(NameStyle.FANCY),
+                owner.getName(NameStyle.FANCY),
+                TextComponent.of("TNT", TextColor.DARK_RED)),
```
Needs to be a translated string, present in the minecraft translations

qlados commented:
I won't have time to work on this small project anymore, which is why I'm going to have to close this PR. If you want to know details contact me via Discord. Sorry!
